### PR TITLE
fix: renaming environments not updated in `secretimports` model

### LINKED ISSUE #920: Environment rename not updating all models
### Describe the bug
When renaming an environment not all references are updated.

Models where the update is missing:

- `folders`
- `folderversions`
- `secretimports` (`imports` + `environment`)
- `secretsnapshots`
- `servicetokendatas` (`scopes`)
- `webhooks`
- maybe more

### To Reproduce
1. Project settings
2. rename any environment (in my case I changed the slug from `prod` to `production`
4. See error (missing folders, imports, etc.)

### Expected behavior
All references to the env beeing updated correctly

### Additional Context
https://github.com/Infisical/infisical/blob/main/backend/src/controllers/v2/environmentController.ts#L132-L214

### PR DESCRIPTION
**Pull request by scomans**
# Description 📣

fixes #920
`secretimports` -> `imports` -> `environment` was missing in #1030 

## Type ✨

- [x] Bug fix

# Tests 🛠️

- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/code-of-conduct). 📝

### CODE REVIEW

File: backend/src/controllers/v2/environmentController.ts [hunk 1]
```diff
@@ -394,6 +394,11 @@ export const renameWorkspaceEnvironment = async (req: Request, res: Response) =>
     { workspace: workspaceId, environment: oldEnvironmentSlug },
     { environment: environmentSlug }
   );
+  await SecretImport.updateMany(
+    { workspace: workspaceId, 'imports.environment': oldEnvironmentSlug },
+    { $set: { 'imports.$[element].environment': environmentSlug } },
+    { arrayFilters: [{ 'element.environment': oldEnvironmentSlug }] },
```
Comment (maidul98) [inline backend/src/controllers/v2/environmentController.ts:400]: I don't think `arrayFilters` is supported by document db ( we have to stay compatible). Can you update the query to make sure that it is compatible https://docs.aws.amazon.com/documentdb/latest/developerguide/mongo-apis.html

Comment (scomans) [reply backend/src/controllers/v2/environmentController.ts:400]: I found the `arrayFilters` in one of Amazon's blog articles about Document DB so I guess it works

https://aws.amazon.com/de/blogs/database/new-amazon-documentdb-features-for-aggregations-arrays-and-indexing/

**Review by maidul98 — APPROVED** [review]
tested and lgtm

**Comment by maidul98** [comment]
tested and lgtm

**Review by maidul98 — APPROVED** [review]
nice work!

**Pull request merged**